### PR TITLE
fix(DataTable) small viewport fixes

### LINKED ISSUE #6540: Pagination overflow on smaller viewports
## Pagination overflow issue

On smaller viewports the pagination overflows its' container.

## What package(s) are you using?

- [x] `carbon-components-react` - 7.15.0

### Expected result

The pagination is visible and if overflowing then scrollable on smaller viewports.

### Actual result

The pagination is overflowing its container.

## Steps to reproduce the issue

1. Include the pagination in a container that is smaller so that it overflows
2. Make sure that you are above 42rem (that is when the overflow is set to visible)

See example here:
https://codesandbox.io/s/carbon-pagination-overflow-issue-zt71k

### Screenshots

Codesandbox:

Screenshot 2020-07-24 at 14 06 06

IBM Cloud:

ibm-cloud-pagination-overflow-bug

### LINKED ISSUE #6544: TableToolbar not full width on smaller viewports
## TableToolbar width issue

On smaller viewports the TableToolbar is not 100% wide.

## What package(s) are you using?

- [x] `carbon-components-react` - 7.16.0

### Expected result

The toolbar should be 100% wide.

### Actual result

The toolbar is not 100% wide.

## Steps to reproduce the issue

1. Use the toolbar and make sure that the table overflows.

### Screenshots

Screenshot 2020-07-24 at 16 06 38

### LINKED ISSUE #6560: DataTable min-width issue
## DataTable min-width issue

On smaller viewports the Data Table enforces a min-width of `31.25 rem` which in some cases makes its container scroll.

## What package(s) are you using?

- [x] `carbon-components-react` - 7.16.0

### Expected result

The DataTable should no extend the container. It should use the container's width and scroll if it's wider.

### Actual result

The DataTable extends the container if it has less than `31.25 rem`.

## Steps to reproduce the issue

1. Use the DataTable in a container and make sure its smaller than `31.25 rem`.

## Additional information

The min-width is enforced [here](https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/components/data-table/_data-table-core.scss#L21). I tried to find more information about the reason for this in the style guide but could not.

### Screenshots

Screenshot 2020-07-27 at 16 42 00

### PR DESCRIPTION
**Pull request by atikenny**
closes #6560, closes #6544, closes #6540 

This PR fixes a couple of small issues when using the `DataTable` and the `Pagination` components on smaller viewports.

#### Changelog

**Changed**

- moved the table to it's own wrapper so that scrolling is applied to the container only, this fixes issues with elements above the table when the table itself overflows
- removed min-width from table container so it doesn't cause double scrolling on pages
- hiding some `Pagination` controls below `md` so that the component is more "mobile friendly" and doesn't overflow causing scrolling

#### Testing / Reviewing

- Review mobile design of `Pagination`
- Regression of `Pagination`
- `DataTable` with additional components should be tested

##### Pagination (mobile)

Screenshot 2020-07-28 at 11 57 17

### CODE REVIEW

File: packages/components/src/components/pagination/_pagination.scss [hunk 1]
```diff
@@ -42,6 +42,19 @@ $css--helpers: true;
     @include carbon--breakpoint('md') {
       overflow: initial;
     }
+
+    // mobile friendly pagination
+    @include carbon--breakpoint-down('md') {
+      .#{$prefix}--pagination__left > *,
+      .#{$prefix}--pagination__right > * {
+        display: none;
```
Comment (tw15egan) [inline packages/components/src/components/pagination/_pagination.scss:50]: Not sure we want to simply hide these items. We'd need to leave the buttons to at least allow navigating to different pages.

File: packages/react/src/components/DataTable/Table.js [hunk 2]
```diff
@@ -36,9 +36,11 @@ export const Table = ({
     [`${prefix}--data-table--visible-overflow-menu`]: !overflowMenuOnHover,
   });
   const table = (
-    <table {...other} className={componentClass}>
-      {children}
-    </table>
+    <div className={`${prefix}--data-table-content`}>
```
Comment (tw15egan) [inline packages/react/src/components/DataTable/Table.js:39]: We may need to look into how the toolbar renders on smaller screens as well...

Screen Shot 2020-07-28 at 10 33 53 AM

File: packages/components/src/components/pagination/_pagination.scss [hunk 3]
```diff
@@ -42,6 +42,19 @@ $css--helpers: true;
...
+      }
+
+      .#{$prefix}--pagination__items-count,
+      .#{$prefix}--pagination__controll-buttons {
```
Comment (tw15egan) [inline packages/components/src/components/pagination/_pagination.scss]: Actually I think the above is fixed with this change: 

```suggestion
      .#{$prefix}--pagination__control-buttons {
```

File: packages/components/src/components/pagination/_pagination.scss [hunk 4]
```diff
@@ -42,6 +42,19 @@ $css--helpers: true;
...
+        display: initial;
```
Comment (tw15egan) [inline packages/components/src/components/pagination/_pagination.scss:55]: Seeing some issues between mobile and full-width

Screen Shot 2020-07-28 at 10 50 34 AM

**Review by tw15egan — COMMENTED** [review]
Ran into a few issues, but the overall functionality seems to be working great

File: packages/react/src/components/DataTable/Table.js (hunk 2 above)
Comment (atikenny) [reply packages/react/src/components/DataTable/Table.js:39]: @tw15egan would you like me to look into that or should I create a separate issue for this? It seems unrelated to the changes I made unless I am mistaken.

Comment (atikenny) [reply packages/react/src/components/DataTable/Table.js:39]: Applied a temporary fix in this PR to make it scroll when there is not enough space. We could do the same trick as in the pagination to hide text that is not needed but that is more of a UX question.

File: packages/components/src/components/pagination/_pagination.scss (hunk 1 above)
Comment (atikenny) [reply packages/components/src/components/pagination/_pagination.scss:50]: I don't think there is too much that can fit into the smallest viewport supported (iPhone, 320 px). I can check how it looks like on that width when the page selector is visible and attach a screenshot.

Comment (atikenny) [reply packages/components/src/components/pagination/_pagination.scss:50]: It actually might just work if I remove the margin on the left, of course would only cover english so we might have issues using other languages.

Screenshot 2020-07-29 at 13 53 49

Comment (atikenny) [reply packages/components/src/components/pagination/_pagination.scss:50]: I was actually wrong, I couldn't make it work when the text is longer

Screenshot 2020-07-29 at 14 21 53

File: packages/react/src/components/DataTable/Table.js (hunk 2 above)
Comment (tw15egan) [reply packages/react/src/components/DataTable/Table.js:39]: This change looks great 👍

File: packages/components/src/components/pagination/_pagination.scss (hunk 1 above)
Comment (tw15egan) [reply packages/components/src/components/pagination/_pagination.scss:50]: Your solution works how I was expecting, using the page buttons and hiding the page selector dropdown on mobile. As long as users have a way to navigate through pages we're good 👍

File: packages/components/src/components/pagination/_pagination.scss [hunk 5]
```diff
@@ -42,6 +42,19 @@ $css--helpers: true;
...
+    // mobile friendly pagination
+    @include carbon--breakpoint-down('md') {
+      .#{$prefix}--pagination__left > *,
+      .#{$prefix}--pagination__right > * {
+        display: none;
+      }
+
+      .#{$prefix}--pagination__items-count,
+      .#{$prefix}--pagination__control-buttons {
+        display: initial;
```
Comment (tw15egan) [inline packages/components/src/components/pagination/_pagination.scss:55]: Wondering if this perhaps needs to be `display: flex`? Caught a small issue on very small breakpoints in Percy. 

Screen Shot 2020-07-29 at 9 45 25 AM


```suggestion
        display: flex;
```

File: packages/react/src/components/DataTable/Table.js (hunk 2 above)
Comment (tw15egan) [reply packages/react/src/components/DataTable/Table.js:39]: Noticed in Percy we're also having this issue with the `with toolbar` version on smaller breakpoints (search gets pushed off the screen to the left). Wondering if it makes sense that container scrolls as well? 

Screen Shot 2020-07-29 at 9 45 55 AM

We can track that in a separate issue if you'd like, but just wanted to point it out if you had a quick fix!

**Comment by tw15egan** [comment]
@atikenny two tiny comments, but otherwise this PR looks great!

File: packages/components/src/components/pagination/_pagination.scss (hunk 5 above)
Comment (atikenny) [reply packages/components/src/components/pagination/_pagination.scss:55]: Interesting, I think this is actually incorrect in Percy. At least this is what I see at 320px.

Screenshot 2020-07-29 at 20 03 20

There is plenty of space on the right hand side of the text, could it be that perhaps the container is 320 px not the pagination component?

**Review by tw15egan — APPROVED** [review]
Thanks for taking the time to fix this! 🎉  👍  ✅

**Review by designertyler — APPROVED** [review]
Looks good! Thank you!

**Review by aledavila — APPROVED** [review]

**Pull request merged**